tests/Controllers/UserGearControllerTest.cs

```diff
@@ -0,0 +1,241 @@
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using ClimbingGearBackend.Controllers;
+using ClimbingGearBackend.Models;
+using ClimbingGearBackend.Interfaces;
+using Xunit;
+using Moq;
+
+namespace ClimbingGearBackend.Tests
+{
+  public class UserGearControllerTest
+  {
+    public List<UserGear> MockUserGearList()
+    {
+      return new List<UserGear>()
+      {
+        new UserGear
+        {
+          UserGearId = 1,
+          GearId = 1,
+          Quantity = 1,
+          Gear = new Gear
+          {
+            Id = 1,
+            Name = "Cam",
+            Description = "Just a cam",
+            Brand = "TestCams"
+          }
+        },
+        new UserGear
+        {
+          UserGearId = 2,
+          GearId = 2,
+          Quantity = 2,
+          Gear = new Gear
+          {
+            Name = "Nut",
+            Description = "Just a nut",
+            Brand = "TestNuts"
+          }
+        },
+        new UserGear
+        {
+          UserGearId = 3,
+          GearId = 3,
+          Quantity = 3,
+          Gear = new Gear
+          {
+            Name = "Carabiner",
+            Description = "Just a Carabiner",
+            Brand = "TestCarabiners"
+          }
+        }
+      };
+    }
+
+    public User MockUser()
+    {
+      return new User
+      {
+        Id = "TestUserId",
+      };
+    }
+
+    public UserGear MockUserGear()
+    {
+      return new UserGear
+      {
+        UserGearId = 1,
+        GearId = 1,
+        Quantity = 1,
+        Gear = new Gear
+        {
+          Id = 1,
+          Name = "Cam",
+          Description = "Just a cam",
+          Brand = "TestCams"
+        }
+      };
+    }
+
+    [Fact]
+    public async Task GetUserGearWithNoId_ShouldReturnAllCurrentUserGear()
+    {
+      var mockRepo = new Mock<IUserGearRepository>();
+      mockRepo.Setup(repo => repo.ListAsync(It.IsAny<ClaimsPrincipal>()).Result)
+        .Returns(MockUserGearList());
+
+      var controller = new UserGearController(mockRepo.Object);
+
+      var result = await controller.GetUserGear();
+
+      var userGear = new List<UserGearDTO>(result.Value);
+      Assert.Equal(3, userGear.Count);
+      Assert.Equal("Cam", userGear[0].Name);
+      Assert.Equal("Nut", userGear[1].Name);
```

Comment:
*snicker* cam nut...

Reply:
One of my tests used to have test data that said post nut...